src/recipe/commands/handlers/uploadRecipeImage.handler.ts

```diff
@@ -0,0 +1,26 @@
+import { CommandHandler, ICommandHandler } from '@nestjs/cqrs';
+import { UpdateRecipeDTO } from '../../../recipe/DTOs/update-recipe.dto';
+import { UploadRecipeImageCommand } from '../../../recipe/commands/impl/uploadRecipeImage.command';
+import { FileService } from '../../../recipe/file.service';
+import { RecipeDAO } from '../../../recipe/recipe.dao';
+
+@CommandHandler(UploadRecipeImageCommand)
+export class UploadRecipeImageHandler
+  implements ICommandHandler<UploadRecipeImageCommand>
+{
+  constructor(private recipeDAO: RecipeDAO, private fileService: FileService) {}
+  async execute(command: UploadRecipeImageCommand) {
+    const { recipeId, file } = command;
+
+    await this.recipeDAO.findRecipeById(recipeId);
+
+    const imageURL = await this.fileService.uploadFileToS3(
+      file.originalname,
+      file.buffer,
+    );
+
+    const recipeWithImage: UpdateRecipeDTO = { imageURL };
+
+    return this.recipeDAO.updateRecipe(recipeId, recipeWithImage);
```

Comment:
have you tried to fetch this recipe to see if the image is available? (having this imageUrl that you save on recipe)

Reply:
What exactly do you mean by this?

Reply:
Are the files that you upload to s3 public? Are you able to download them as a client using imageURL?